src/search.rs

```diff
@@ -571,4 +571,39 @@ mod tests {
         assert_eq!(results.hits.len(), 1);
         Ok(())
     }
+
+    #[meilisearch_test]
+    async fn test_generate_tenant_token_from_client(client: Client, index: Index) -> Result<(), Error> {
+        use crate::key::{KeyBuilder, Action};
+
+        setup_test_index(&client, &index).await?;
+
+        let key = KeyBuilder::new("key for generate_tenant_token test")
+            .with_action(Action::All)
+            .with_index("*")
+            .create(&client).await.unwrap();
+        let allowed_client = Client::new("http://localhost:7700", key.key);
+
+        let search_rules = vec![
+            json!({'*': {}}),
+            json!({'*': Value::Null }),
+            json!(['*']),
+            json!({ "*": { "filter": "kind = text" } }),
```

Comment:
Is the choice of different type of quotes expected?

Reply:
No it's not! :p